force-app/main/default/classes/ProgramService.cls

```diff
@@ -22,4 +25,12 @@ public with sharing class ProgramService {
         }
         return programNamesById;
     }
+
+    public List<ProgramCohort__c> getCohortsForProgram(Id programId) {
+        Set<String> fields = new Set<String>{
```

Comment:
Interesting - I have no issue with making that change, but I feel like it makes the method in the Selector usage-specific ... as in, we need these two fields specifically for assembling the dropdown options, but other use case will probably need a different selector method that gathers a different group of fields. As a result, I would be inclined to rename the selector method to something specific for that use case, perhaps "getCohortsByProgramForSelectOptions" or something? That's wordy, but specific?

Reply:
Well even if we needed to add fields for another method we would still have what we needed to create the options. I think we just add the fields as we need them.

Reply:
Actually ... I was looking for other examples and ProgramEngagementSelector has examples of both passing the field list in as an argument (`getProgramEngagementsByProgramId(Id programId, Set<String> fields, Set<String> stages)`) and baking the field list into the implementation (`getProgramCohortsByProgramId(Id programId)`). 

Furthermore ... getProgramCohortsByProgramId() actually looks like it's actually delivering the same behavior as this method. Should we try and consolidate on one or the other rather than duplicating that behavior? Is it worth looking into whether getProgramCohortsByProgramId belongs in the ProgramEngagementSelector?

Reply:
Great catch, yea if we already have that method in there, we should reuse it.